gcs-fetcher/pkg/fetcher/fetcher.go

```diff
@@ -298,6 +317,17 @@ func (gf *Fetcher) fetchObjectOnce(ctx context.Context, j job, dest string, brea
 
 	r, err := gf.GCS.NewReader(ctx, j.bucket, j.object)
 	if err != nil {
+		// Check for AccessDenied failure here and return a useful error message on Stderr and exit immediately.
```

Comment:
It seems like errors from `NewReader` should be `*googleapi.Error`s, is that not what you've seen in your tests?

https://godoc.org/cloud.google.com/go/storage#hdr-Errors

I thought you should be able to do:

```go
r, err := gf.GCS.NewReader(...)
if err != nil {
  if gerr, ok := err.(*googleapi.Error); ok && gerr.Code == http.StatusForbidden {
    // got 403
  } else {
    // some other error
  }
}
```

If that's not what you're seeing though then maybe GCS has some other semantic difference between other GCP APIs.

Reply:
I tried casting to (googleapi.Error) which wouldn't compile. I didn't try (*googleapi.Error) though. I will give it a go.

Reply:
Thanks to your help, I got this to work.